### PR TITLE
Update edit command

### LINKED ISSUE #7: As an expert user, I can edit my students' personal details
... so that I can update outdated data.

### PR DESCRIPTION
**Pull request by VaishakAnand**
Closes #7 

Changes made: 

- Updated Edit functionality for new Student and Admin classes.
- Updated existing test cases.
- Unit tests for EditCommandParserTest for Admin fields
- Unit tests for EditCommandTest for Admin fields

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -83,18 +107,29 @@ public CommandResult execute(Model model) throws CommandException {
...
+        ClassVenue updatedClassVenue = editAdminDescriptor.getClassVenue()
+                .orElse(studentToEdit.getAdmin().getClassVenue());
+        Fee updatedFee = editAdminDescriptor.getFee()
+                .orElse(studentToEdit.getAdmin().getFee());
+        PaymentDate updatedPaymentDate = editAdminDescriptor.getPaymentDate()
+                .orElse(studentToEdit.getAdmin().getPaymentDate());
+        Set<AdditionalDetail> updatedAdditionalDetails = editAdminDescriptor.getAdditionalDetails()
+                .orElse(studentToEdit.getAdmin().getDetails());
+        Admin updatedAdmin = new Admin(updatedClassVenue, updatedClassTime, updatedFee, updatedPaymentDate,
+                updatedAdditionalDetails);
```
Comment (ong-yinggao98) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: Looks good! Perhaps one suggestion would be to call `editAdminDescriptor::isAnyFieldEdited` so we can just create the new student using `student.toEdit().getAdmin()`.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -207,8 +232,109 @@ public boolean equals(Object other) {
...
+        }
+
+        /**
+         * Returns an unmodifiable detail set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code details} is null.
+         */
+        public Optional<Set<AdditionalDetail>> getAdditionalDetails() {
+            return (details != null) ? Optional.of(Collections.unmodifiableSet(details)) : Optional.empty();
+        }
```
Comment (ong-yinggao98) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: Really damn minor note but you could probably just use `Optional.ofNullable(details).map(Collections::unmodifiableSet)`

**Review by ong-yinggao98 — APPROVED** [review]
Looks good! I've left some comments if you are interested

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 1 above)
Comment (VaishakAnand) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Good suggestion! I've implemented it in my latest commit.

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 2 above)
Comment (VaishakAnand) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Thanks again, I've just implemented this in my latest commit.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -207,8 +237,110 @@ public boolean equals(Object other) {
...
+
+        /**
+         * Returns an unmodifiable detail set, which throws {@code UnsupportedOperationException}
+         * if modification is attempted.
+         * Returns {@code Optional#empty()} if {@code details} is null.
+         */
+        public Optional<Set<AdditionalDetail>> getAdditionalDetails() {
+            return Optional.ofNullable(details).map(Collections::unmodifiableSet);
+            // return (details != null) ? Optional.of(Collections.unmodifiableSet(details)) : Optional.empty();
+        }
```
Comment (AlexCQY) [inline src/main/java/seedu/address/logic/commands/EditCommand.java]: Just to clarify for now we cant edit the additional details?

**Review by AlexCQY — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/EditCommand.java (hunk 3 above)
Comment (VaishakAnand) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: Correct, we can't edit for now. I can do a separate PR to allow for such an edit

Comment (AlexCQY) [reply src/main/java/seedu/address/logic/commands/EditCommand.java]: okay thanks

**Review by hogantan — APPROVED** [review]
LGTM

**Review by csiongn — APPROVED** [review]
LGTM

**Comment by csiongn** [comment]
I think need update the CommandTestUtil file to this: https://github.com/AY2021S1-CS2103T-W15-2/tp/blob/master/src/test/java/seedu/address/logic/commands/CommandTestUtil.java

**Pull request merged**